linera-witty-macros/src/wit_export/function_information.rs

```diff
@@ -216,7 +216,7 @@ impl<'input> FunctionInformation<'input> {
     }
 
     /// Generates the code to export a host function using a mock Wasm instance for testing.
-    #[cfg(feature = "mock-instance")]
+    #[cfg(with_testing)]
```

Comment:
I wonder if we couldn't make this a trait?

I generally think that when you have multiple feature flags, it's helpful to put them in different trait instances that can be abstracted into a module.  Then, the whole module can be conditionally-compiled out, and it gets much simpler to reason about — since the presence of the trait instance in the module guarantees that the conditionally-compiled code satisfies the interface you expect it to satisfy (which otherwise wouldn't be discovered until the use site, potentially far away in the codebase).

Reply:
That's fair, could we make it an issue so we don't forget about it?